### ISSUE TITLE
Incorrect FirewallException.IcmpTypesAndCodes property type

### ISSUE THREAD

**Issue by oleg-shilo** [oleg-shilo/wixsharp#1606]
### Discussed in https://github.com/oleg-shilo/wixsharp/discussions/1603

<div type='discussions-op-text'>

<sup>Originally posted by **dbieber-rfideas** August  8, 2024</sup>
Hello,

Thanks for the work on WixSharp. It's a great project.

I'm having trouble figuring out how to add an in/out firewall rule. I have successfully added firewall rules, but it only creates an inbound rule. I need to be able to configure both inbound and outbound type rules. Is that possible with WixSharp? It is supported by Wix4 itself, according to their Schema documentation:
[FirewallException](https://wixtoolset.org/docs/schema/firewall/firewallexception/
)
`Outbound (wxs:YesNoTypeUnion) : If "yes", registers an outbound firewall rule. The default is "no".`

. . . 

Is there a way to define the rule as inbound/outbound, or just both?</div>

---

The issue is caused by the WIX XML that WixSharp generates containing an inapplicable attribute:

<Component Id="_...Reader20Configuration20Utility___" Guid="7521944c-4f42-4aa8-97e7-5338c6c1878e" KeyPath="yes">
            <FirewallException Id="_...Reader20Configuration20Utility___" Name="Configuration Utility" Port="80" Enabled="yes" IcmpTypesAndCodes="no" Outbound="no" xmlns="http://wixtoolset.org/schemas/v4/wxs/firewall" />

            <CreateFolder />
            <RemoveFolder Id="INSTALLDIR" On="uninstall" />
          </Component>

The IcmpTypesAndCodes="no" attribute causes the installer to fail to install the rule. If I remove that attribute, the rule works. If I add it back (with both "no" and "yes" values), it fails. I think this is because that property only applies to rules that are using the ICMPv4 protocol. Since this rule is TCP, it doesn't make sense (if you look at the windows firewall advanced GUI, you will see that this setting is grayed out unless ICMPv4 protocol is selected).

Is it possible for WixSharp to conditionally insert this attribute only when the ICMPv4 protocol is selected (or just omit the attribute since WixSharp doesn't seem to support ICMPv4 protocol)?ICMPv4 protocol is selected

**Comment by oleg-shilo** [comment]
Oh, sorry, it was my misunderstanding.

I interpreted your description "It is caused by the WIX XML that WixSharp generates containing an inapplicable attribute:" a fact that WixSharp calculates and inserts this attribute by itself. I was wrong.

I checked the solution and this WiX4 attribute is directly mapped to the `FirewallException.IcmpTypesAndCodes` property thus WixSharp has no business of modifying it but rather taking the user-specified value or using the default.

Though, there is a mistake in the default value encoded in WixSharp. Instead of being declared as a string, it is declared as a bool. That's why the default attribute value is `no`.

Fixed. Will be available in the very next release (or sooner as a pre-release).
